src/sqllogictest/src/runner.rs

```diff
@@ -1122,6 +1132,21 @@ impl RunnerInner {
             }
         }
 
+        match output {
+            Ok(_) => {
+                if !*in_transaction {
```

Comment:
Can this run `transactions.slt` without modification? You may need to make this a command-line option after all, this way people will not be suprised that sqllogictest is running SQL statements that are not present in the original SLT file.